src/main/java/org/zalando/nakadi/validation/SchemaEvolutionService.java

```diff
@@ -97,6 +102,12 @@ public EventType evolve(final EventType original, final EventTypeBase eventType)
 
         final Version.Level changeLevel = semanticOfChange(changes, original.getCompatibilityMode());
 
+        if (changeLevel == NO_CHANGES && !original.getSchema().getSchema().equals(eventType.getSchema().getSchema())) {
+            LOG.error("undetected schema changes from {} to {}", original.getSchema().getSchema(),
+                    eventType.getSchema().getSchema());
+            throw new MyNakadiRuntimeException1("undetected schema changes");
```

Comment:
I agree with you in case we needed to handle this exception. But in this situation, this is really like a NullPointer. Something unexpected happened: something changed in the schema but our algorithm for comparison was unable to precisely tell why. So we just 💣 forcefully to halt all further operations.

Later, we check the logs, see what the input looked like, fix the bug and tell the user to try again.